### PR TITLE
[PERF-299] Add sample script using TestNG framework

### PR DESCRIPTION
**Pull request by farhan-sauce**
Ticket: [PERF-299](https://saucedev.atlassian.net/browse/PERF-299)

### CODE REVIEW

File: TestNG/src/test/java/com/yourcompany/Pages/SauceDemoPage.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    public void sleep(long msecs) {
+        try {
+            Thread.sleep(msecs);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Map getPeroformance() {
```
Comment (yfangsl) [inline TestNG/src/test/java/com/yourcompany/Pages/SauceDemoPage.java]: getPerformance

File: TestNG/src/test/java/com/yourcompany/Tests/PerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        this.createDriver(browser, version, os, method.getName());
+        WebDriver driver = this.getWebDriver();
+
+        this.annotate("Visiting SauceDemo page...");
+        SauceDemoPage page = SauceDemoPage.visitPage(driver);
+
+        page.loginUser();
+        page.visitPage("/inventory.html");
+        page.sleep(2000);
+        Map<String, Object> performance = page.getPeroformance();
```
Comment (yfangsl) [inline TestNG/src/test/java/com/yourcompany/Tests/PerformanceTest.java]: getPerformance

**Review by yfangsl — CHANGES_REQUESTED** [review]
So while this generally seems fine as a demonstration of how to use custom commands, I'm not sure this is what the story was actually about, and I'm not sure it can be properly finished at the moment.  I think the story was about the *new* performance command that uses the baseline metrics.  So more based on the examples here: https://wiki.saucelabs.com/display/DEVTOOLS/Automated+Performance+Assertion

**Comment by farhan-sauce** [comment]
@yfangs Did we deploy `baseline metrics` custom command yet? I thought [this ticket](https://saucedev.atlassian.net/browse/PERF-291) is still a blocker and we decided to add baseline metrics assertion later once we deploy it.

I would be happy to use new performance command if it is already deployed and there is no blocker.

**Comment by seanenright** [comment]
@farhan-sauce @yfangsl merging so we can close the ticket (per christine).
there is a follow up ticket to update each of the sample scripts with the new custom command 
https://saucedev.atlassian.net/browse/PERF-353

**Pull request merged**